webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/GroupsApi.scala

```diff
@@ -1367,61 +1369,68 @@ class GroupApiService14(
-                                      .checkValueTypes(matchingChildProperties.map(NodePropertyHierarchy(p.prop, _)))
-                                      .isLeft
-                                    (
-                                      p,
-                                      matchingChildProperties.flatten,
-                                      hasConflicts
-                                    )
-                                  })
-                              })
-                              .toIO
-                          }
+      params               <- paramRepo.getAllGlobalParameters().map(_.map(p => (p.name, p)).toMap)
+      parentProperties      = MergeNodeProperties.forGroup(group, allGroups, params)
+      checkedNodeProperties = {
+        // to check for potential type conflicts in the properties of node, we need to look at each node
+        nodes.map { nodeFact =>
+          // for each property, find merged properties for nodes in the group
+          val resolvedProperties = MergeNodeProperties
+            .forNode(
+              nodeFact,
+              groupLibrary.getGroupTarget(nodeFact).values,
+              params
+            )
+            .prependFailureMessage(
+              s"Inherited properties are in an error state when searching properties in relation with group ${allGroups
+                  .get(groupId)
+                  .map(_.nodeGroup.name)
+                  .getOrElse("current group")} (with ID '${groupId.serialize}') and all its nodes: "
+            )
+          // for each property of the group, search all properties in node properties
+          // and take them into account for property status
+          val childProperties    = resolvedProperties.resolved
+          val success            = parentProperties.resolved.map { p =>
+            // a single node prop should be matching the current one
+            // and we should validate the found the hierarchy of the child node property found
+            val matchingChildProperties = childProperties.collectFirst {
+              case cp if cp.prop.name == p.prop.name => cp.hierarchy
+            }
+            InheritedPropertyStatus.fromChildren(
+              p,
+              matchingChildProperties
+            )
+          }
+          val error              = resolvedProperties match {
+            case f: FailedNodePropertyHierarchy  => PropertyStatus.fromFailedHierarchy(f)
+            case s: SuccessNodePropertyHierarchy => Chunk.empty
+          }
 
+          success ++ error
+        }
+      }
+      groupPropertyStatuses = checkedNodeProperties match {
+                                case Nil       => // no Node is group : just use resolved group properties
+                                  parentProperties match {
+                                    case s: SuccessNodePropertyHierarchy =>
+                                      s.resolved.map(InheritedPropertyStatus.from(_))
```

Comment:
I just realized we should check that the resolved property is "Displayed" before transforming it in 8.2 (the "Visibility" is only from 8.2 so it simplifies things here)